sycl/include/sycl/ext/oneapi/experimental/work_group_memory.hpp

```diff
@@ -36,29 +36,52 @@ class work_group_memory_impl {
 } // namespace detail
 namespace ext::oneapi::experimental {
 
+struct indeterminate_t {};
+inline constexpr indeterminate_t indeterminate;
```

Comment:
Ah yes, this makes total sense. I still think I'd rather change the name of the PR rather than have it as a separate patch because otherwise I'd need this PR to have the old version of the tests without `sycl::indeterminate` and then once the PR adding it is merged, I'd have to also change these tests again.